src/common/util.js

```diff
@@ -333,6 +333,8 @@ const PRECEDENCE = {};
   ["+", "-"],
   ["*", "/", "%"],
   ["**"],
+  // TSAsExpression
+  ["as"],
```

Comment:
I put a precedence of TSAsExpression to resolve https://github.com/prettier/prettier/pull/7869/files#r398206153, I'm not sure if this is correct.